frontends/web/src/api/devicessync.ts

```diff
@@ -77,3 +79,24 @@ export const attestationCheckDone = (
   });
   return unsubscribe;
 };
+
+
+export const signProgress = (
+  cb: (meta: TSignProgress) => void
+): TUnsubscribe => {
+  return apiWebsocket(payload => {
+    if ('type' in payload && payload.type === 'device' && payload.data === 'signProgress') {
+      cb(payload.meta);
+    }
+  });
+};
+
+export const signConfirm = (
+  cb: (signConfirm: boolean) => void
```

Comment:
I think the backend doesn't return anything here, but maybe check if there is something in the event object, or check this handler in the backend go implementation.

so I thin you can just use `() => void` to use this as a callback.

and then change

```js
  signConfirm(() =>
    this.setState({ signConfirm: true })
  ),
```